pkg/apis/naiserator/v1alpha1/types.go

```diff
@@ -168,11 +168,13 @@ func (in Application) Hash() (string, error) {
 	// struct including the relevant fields for
 	// creating a hash of an Application object
 	relevantValues := struct {
-		AppSpec ApplicationSpec
-		Labels  map[string]string
+		AppSpec     ApplicationSpec
+		Labels      map[string]string
+		Annotations map[string]string
```

Comment:
I'm afraid this will create an impossible situation. Naiserator stores the aforementioned hash value in the annotations. Thus, annotations cannot be included here.

Reply:
https://github.com/nais/naiserator/commit/8de73ea1dbb09f95f23ffb0f16e8a6baf087a17f
maybe?

Reply:
Though, what is the point of adding this here? Updating the annotation will force a re-synchronization of the resource. Is this what you want?

Reply:
https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#checking-rollout-history-of-a-deployment
To support this ^. When the application spec is annotated with this specific annotation it would be nice to annotate the deployment resource as well. Annotating the deployment resource does not create a new replicaset.

This is just a suggestion, not at all well though trough, just thought id put my thoughts into code to see if it was easy to implement.

Any drawbacks to triggering a re-sync?

Reply:
We could still support this without re-sync, but then people would have to make sure they include the annotation every time they `k apply -f app.yaml`, which might be fine.